src/config.ts

```diff
@@ -166,6 +178,7 @@ export interface Config {
  * user-provided value will be used to extend the default value.
  */
 export const defaultConfig = {
+  clsMechanism: 'auto' as ContextPropagationMechanism,
```

Comment:
I see. We lost type-safety from it though. If someone makes a typo here in the future, it may not be detected by the compiler.

Would it be helpful to use `Partial`? I.e. make `Config`'s field required and define `PartialConfig` out of it using `Partial`... Just a random idea. Feel free to ignore.